src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -337,67 +337,23 @@ internal static void SaveStreamToFile(Stream stream, string filePath, PSCmdlet c
             WriteToStream(stream, output, cmdlet, contentLength, cancellationToken);
         }
 
-        private static string StreamToString(Stream stream, Encoding encoding, CancellationToken cancellationToken)
+        private static async Task<Encoding> GetStreamEncodingAsync(Stream stream, string? characterSet, CancellationToken cancellationToken)
         {
-            StringBuilder result = new(capacity: ChunkSize);
-            Decoder decoder = encoding.GetDecoder();
+            bool isDefaultEncoding = !TryGetEncodingFromCharset(characterSet, out Encoding encoding);
```

Comment:
The precedence of encoding detection is unclear to me.
By looking at the code, when `characterSet` is specified, `isDefaultEncoding` will be `false`, and
- meta element will not be checked
- encoding detected from BOM will take precedence, then the encoding resolved from `characterSet`

when `characterSet` is not specified, `isDefaultEncoding` will be `true`, and
- meta element will be checked
- character set from the meta element will take precedence, then the encoding detected from BOM, and then the default encoding

As is shown, the encoding detection is inconsistent -- if the `characterSet` specified in header is lower precedence than BOM, then why the `characterSet` specified in meta element is higher than BOM?

Can you please summarize the precedence you use and the reason for that (_e.g. to avoid breaking change? RFC defined? If they are contradicted with each other, then maybe we should make a breaking change to adhere to RFC?_) The summarization needs to be put in the code as comment.

Reply:
> Can you please summarize the precedence you use and the reason for that (e.g. to avoid breaking change? RFC defined? If they are contradicted with each other, then maybe we should make a breaking change to adhere to RFC?) The summarization needs to be put in the code as comment.

@CarloToso I see that you've made changes to the encoding detection. But can you please summarize the precedence and the reason for using that precedence?